localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_api_gateway.py

```diff
@@ -174,6 +174,11 @@ def _headers_of(parameters: TaskParameters) -> Optional[dict]:
                 for forbidden_prefix in StateTaskServiceApiGateway._FORBIDDEN_HTTP_HEADERS_PREFIX:
                     if key.startswith(forbidden_prefix):
                         raise ValueError(f"The 'Headers' field contains unsupported values: {key}")
+
+                value = headers.get(key)
+                if isinstance(value, list):
+                    headers[key] = f"[{",".join(value)}]"
```

Comment:
That's an embarrasing error. Went for lunch after pushing this rather than running tests locally first. 😅
Github should really add an option to draft revisions
Pushed the fix now.

Reply:
No worries 😺 
We were missing instructions about pre-commit hooks in the contributing guide to catch such errors early. Hope this helps 🚀  https://github.com/localstack/localstack/pull/11738/files#diff-829d1c136d14d3a54c532b14af632e300b2401bbe8776be880ab61a13a4e29ccR116